VDA-SBT/contracts/SoulboundNFT.sol

```diff
@@ -190,57 +164,58 @@ contract SoulboundNFT is VDAVerificationContract,
 
     }
 
-
     /**
      * @dev See {IsoulboundNFT}
      */
     function claimSBT(
         address did,
-        string calldata sbtType,
-        string calldata uniqueId,
-        string calldata sbtURI,
-        address recipient,
-        bytes calldata signature,
-        bytes calldata proof
+        SBTInfo calldata sbtInfo,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
     ) external override returns(uint) {
-
-        require(isValidSBTType(sbtType), "Invalid SBT type");
+        require(isValidSBTType(sbtInfo.sbtType), "Invalid SBT type");
         {
             bytes memory params = abi.encodePacked(
                 did,
-                "-",
-                sbtType,
-                "-",
-                uniqueId,
-                "-");
-            params = abi.encodePacked(
-                params,
-                sbtURI,
-                "-",
-                recipient,
-                "-");
+                // "-",
+                sbtInfo.sbtType,
+                // "-",
+                sbtInfo.uniqueId,
+                // "-",
+                sbtInfo.sbtURI,
+                // "-",
+                sbtInfo.recipient);
+
             params = abi.encodePacked(
```

Comment:
Wrapping the `params` up twice is very confusing and seems unnecessary. I found this very difficult to follow.

Reply:
This is because, "stack too deep" issue in the solidity.
Can't encode all the parameters with one `encodePacked()` function.

Reply:
That is the limit of Solidity.

Reply:
Oh :(

Where can I read more about that limit?